datanode-ui/src/libs/types/api/EnvironmentInterface.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ *
+ * Copyright 2023 Odysseus Data Services, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import { DescriptorInterface } from "@/libs/types";
+
+export interface EnvironmentInterface {
```

Comment:
If we are mixing two interfaces (Docker vs isolated Runtime Environments with a Descriptor) we should clearly specify how to cope with this in the future

Reply:
It is true that we need a strategy for proper generalization on top of docker and tarball modes, but these interfaces in frontend code are quite ephemeral. And since both backend and frontend are in the same repository, later adjustments have minimal overhead in this area.